### ISSUE TITLE
Redirect doesn't work with "$currentLocale" based on "id"

### ISSUE THREAD

**Issue by jsunsawyer** [sjelfull/craft-countryredirect#7]
In `CountryRedirectService.php` the function `getCountryCode()` sets `$currentLocale` based on an `id` rather than `handle`.

`$currentLocale = Craft::$app->getSites()->currentSite->id;`

This means that the following will never return true:

```
foreach ($countryMap as $countryCode => $locale) {
    if ($locale === $currentLocale) {
        // Override if parameter is in there
        $this->setCountryCookie($countryCode);
    }
}
```

Changing `id` to `handle` resolves the issue for everything except for our region based sites.

Here's our country map from the config:

```
'countryMap' => [
  'US' => 'en_us', // US (English)
  'GB' => 'en_gb', // UK (English)
  'CA' => 'en_ca', // Canada (English)
  'NZ' => 'en_nz', // New Zealand (English)
  'FR' => 'fr_fr', // France (French)
  'CH' => [
    'fr' => 'fr_ch', // Switzerland (French)
    'de' => 'de_ch', // Switzerland (German)
    'it' => 'it_ch', // Switzerland (Italian)
  ],
  'DE' => 'de_de', // Germany (German)
  'BR' => 'pt_br', // Brazil (Portuguese)
  '*' => 'en_us',
],
```

Should redirects be working without this change? And how can we get our region based sites to work properly?

**Comment by sjelfull** [comment]
Thanks for reporting this. Will look into this once I'm back from vacation next week.

**Comment by jsunsawyer** [comment]
@sjelfull Just wanted to check in on the status of this issue. We're expecting to launch a big site update that depends on this plugin this month.

**Comment by sjelfull** [comment]
Taking a stab at this within the next week.

**Comment by sjelfull** [comment]
This change does indeed fix the redirect.

I was also able to verify that region matching works.

The steps I took:

1. Setup config with a map of regions in `CH` and test IP that matches somewhere in Switzerland: 
```php
<?php

return [
    'enabled'       => true,
    'enableLogging' => true,
    'countryMap'    => [
        'NO' => 'dutchSite',
        'CH' => [
            'fr' => 'foreignSiteBranch', // Switzerland (French)
            'de' => 'dutchSite', // Switzerland (German)
            'it' => 'dutchSite', // Switzerland (Italian)
        ],
    ],
    'overrideIp'    => '31.3.128.0',
];
```

2. Made sure French is in the top of my accepted languages:

![image](https://user-images.githubusercontent.com/10508/44089551-77eacfae-9fc7-11e8-87de-9b34dd154536.png)

3. Visit test site
4. Result: I was redirected to the site matching the `foreignSiteBranch` handle (test.dev/foreign).

Could you explain in more detail the steps you took to test region matching, and what results you are seeing?

**Comment by jsunsawyer** [comment]
We're still unable to directly visit the nested Switzerland (CH) sites with your plugin enabled.

After the steps you listed, are you then able to switch to a different language within Switzerland via a direct nav link?

**Our URL structure:**
Single languages: `example.test/de`
Regional languages: `example.test/ch/it`

**Detailed Steps:**
1. Set `overrideIp` to `31.3.128.0`
2. Clear cookies
3. Visit site
    - Redirects to Switzerland (French) `/ch/fr`
    - Cookie gets set to `CH`
4. Attempt to visit a different Switzerland site
    - Click URL `http://example.test/ch/de/?selected-locale=%E2%9C%93`
    - Site stays at French – Switzerland `/ch/fr`
5. Attempt to visit a single language site
    - Click URL `http://example.test/fr/?selected-locale=%E2%9C%93`
    - Site loads the clicked URL
    - Cookie gets set to `FR`
6. Attempt to visit Switzerland (French)
    - Click URL `http://example.test/ch/fr/?selected-locale=%E2%9C%93`
    - Site stays at French URL `http://test.test/fr/?redirected=%E2%9C%93`
    - Cookie stays `FR`

**Comment by sjelfull** [comment]
Aha - I realise I never verified if the override worked on sites that use regions. If that is the case, I will release a fix soon.

**Comment by gilleard** [comment]
Yeah, as is the override still doesn't work with regional languages - the equality check in getCountryCode between the current site handle and the array obviously fails.
